### PR TITLE
Add SyntheticDataset and SyntheticDatasetConfig to diarizers

### PR DESCRIPTION
**Pull request by kamilakesbi**
# What does this PR do?

- Adds `SyntheticDataset` and `SyntheticDatasetConfig` classes to the `diarizers` package. 

- Updates the package. 

- Adds a `datasets/synthetic_dataset.py` script to assist in creating synthetic datasets. 

- Updates `datasets/README.md` to explain how to use the synthetic dataset. 

- Add scripts to push the Sakura and Callfriend datasets to the hub.

cc @sanchit-gandhi

### CODE REVIEW

File: datasets/README.md [hunk 1]
```diff
@@ -108,3 +105,66 @@ python3 spd_datasets.py \
     --hub_repository=diarizers-community/callhome \
 ```
 
+
+## 2. Generate a synthetic dataset compatible with diarizers:
```
Comment (sanchit-gandhi) [inline datasets/README.md:109]: Would be great to explain here why you'd want to use synthetic data (over real data) and how it scales up to an arbitrary amount of training data.

File: datasets/README.md [hunk 2]
```diff
@@ -108,3 +105,66 @@ python3 spd_datasets.py \
...
+    --silence_proba=3 \
+    --denoise=False \
+    --bn_path=None \
+    --ir_path=None \
+    --num_proc=2 \
+    --push_to_hub=True \
+    --hub_repository='test'
+```
+
+Find more informations on how to use 🤗 Diarizers synthetic speaker diarization pipeline in this notebook: [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1kaKv5Qa2dUuEwyLoFeh5mCwgy8O_ZdYA#scrollTo=27RrvTZte4BF).
```
Comment (sanchit-gandhi) [inline datasets/README.md:170]: Shall we complete the example here and show how you then train on this synthetic dataset? Such that we can open the discussion on the benefits/drawbacks of using synthetic data?

File: datasets/spd_datasets.py [hunk 3]
```diff
@@ -3,11 +3,18 @@
 import glob
 import os
 
+from moviepy.editor import AudioFileClip
```
Comment (sanchit-gandhi) [inline datasets/spd_datasets.py:6]: Is this a new dependency that we require?

File: datasets/synthetic_dataset.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
+from dataclasses import dataclass, field
+
+import torch
+from transformers import HfArgumentParser
+
+from diarizers import SyntheticDataset, SyntheticDatasetConfig
+
+
+@dataclass
+class ASRDatasetArguments:
```
Comment (sanchit-gandhi) [inline datasets/synthetic_dataset.py:10]: Wondering whether we need both `ASRDatasetArguments` **and** `SyntheticDatasetConfig`? IMO there's some unnecessary redundancy here -> we only need one of the two config classes.

I would probably stick with `SyntheticDatasetConfig` to make it easy, and have users define the args in a python cell:
```python
from diarizers import SyntheticDatasetConfig, SyntheticDataset

synthetic_config = SyntheticDatasetConfig(
        dataset_name=...,
        subset=...,
        split=...,
...
)

dataset = SyntheticDataset(synthetic_config).generate(batch_size=..., num_proc=...)
```

File: datasets/README.md (hunk 2 above)
Comment (sanchit-gandhi) [inline datasets/README.md:170]: Unfortunately I don't have access to the notebook!

**Review by sanchit-gandhi — COMMENTED** [review]
Would be great to add an end-to-end example showing how to use this dataset to train a model, explaining the benefits/drawbacks compared to in-distribution data

Otherwise the pipeline code itself looks very clean. Linking the notion page used for the initial review here: https://www.notion.so/huggingface2/Synthetic-Diarization-Pipeline-Review-82241c0c58a544ca91247e5371c892b5?pvs=4

File: datasets/README.md (hunk 2 above)
Comment (kamilakesbi) [reply datasets/README.md:170]: Just shared!

Comment (kamilakesbi) [reply datasets/README.md:170]: We could do this but IMO the impact of such discussion would be quite limited. For now, I would only introduce the pipeline as done in the notebook. WDYT ?

File: datasets/synthetic_dataset.py (hunk 4 above)
Comment (kamilakesbi) [reply datasets/synthetic_dataset.py:10]: If I understand well you propose to remove the `synthetic_pipeline.py` script from the repository and propose instead a python code snippet using `SyntheticDatasetConfig` in the readme ?

**Pull request merged**